packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -108,9 +114,25 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         sharedObjectRegistry: ISharedObjectRegistry,
         existing: boolean,
     ): FluidDataStoreRuntime {
-        return new FluidDataStoreRuntime(context, sharedObjectRegistry, existing);
+        return new FluidDataStoreRuntime(
+            context,
+            sharedObjectRegistry,
+            existing,
+            async (rt) => requestFluidObject(rt, "/"));
     }
 
+    /**
+     * Fluid handle to the data store runtime. It'll only be defined if initializeEntrypoint was passed to the
+     * constructor when instantiating the runtime. Exposed so we can start making handles a first-class citizen and
+     * the primary way of interacting with some fluid objects. Use it if possible, but for now all code paths should
+     * first check if it's not undefined, and keep using current approaches (e.g. using the request pattern) if it is.
+     */
+    public get IFluidHandle() {
```

Comment:
I would expect this to always be defined?  Why not create a handle in all cases?

Reply:
eventually we will, but since initializeEntrypoint must be optional to start, we can't know we can create a handle

Reply:
@anthony-murphy doesn't this connect with your other comment, that we could initialize the handle with the request pattern if `initializeEntrypoint` wasn't provided?

Even if someone was subclassing `FluidDataStoreRuntime` and took this change in without providing `initializeEntrypoint`, we'd still be initializing the handle.

Reply:
i think providing a default implementation is a bad idea due to the potential for reentrancy and infinite recursion